### ISSUE TITLE
plot.section should allow plotting of any field contained in @data

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#818]
`plot.section()` has `which=` values for a range of fields, but if a field exists in an object that isn't contained in that list it's not possible to make a section plot of it. 

One approach is to add more `which=` values for different data types.

Another would be to work similarly to `plotProfile()` where it checks the `xtype=` argument against the names of the `@data` fields and makes a plot if one of them matches.

**Comment by dankelley** [comment]
Question for @richardsc -- can you see why there should be these whacky pressures in the last two non-NA values in one of the argo profiles?

```r
> argo@data$pressure[,14]
 [1]    4.3    9.6   14.3   19.2   24.0   29.1   34.4   39.1   49.3   59.5   69.5   79.3   89.0   99.4  109.1  119.1  129.4  139.0  149.1  159.4  169.5
[22]  179.5  189.2  199.1  219.2  239.6  259.6  279.1  299.1  329.0  359.3  399.4  448.7  499.2  549.5  599.1  649.2  699.3  848.8  899.2  949.4  999.1
[43] 1049.2 1099.3 1149.1 1198.9 1248.9 1299.5 1349.5 1383.0 3411.6 6178.5     NA     NA     NA     NA
```

I ran across this because I noticed that section plots were drawn down to 6km depth. So I did another test
```r
> sum(argo@data$pressure>2000,na.rm=TRUE)
[1] 5
```
and so there are a few more problems in other profiles too.

I'm inclined to just find anything that is over 2km depth and set it to NA. But maybe the data mean something..?? Any ideas or suggestions?

**Comment by dankelley** [comment]
A bit more info:
```r
> stem(argo@data$pressure)

  The decimal point is 3 digit(s) to the right of the |

  0 | 00000000000000000000000000000000000000000000000000000000000000000000+6748
  0 | 55555555555555555555555555555555555555555555555555555555555555555555+1988
  1 | 00000000000000000000000000000000000000000000000000000000000000000000+1764
  1 | 55555555555555555555555555555555555555555555555555555555555555555555+677
  2 | 
  2 | 
  3 | 4
  3 | 
  4 | 
  4 | 
  5 | 
  5 | 
  6 | 2
  6 | 555
```

**Comment by dankelley** [comment]
Hm. I'm beginning to think I need to put flags into `read.argo()`. Manually, doing
```r
f <- nc_open("/Users/kelley/src/oce/create_data/argo/6900388_prof.nc")
pflag <- ncvar_get(f, "PRES_QC")
pflag
```
shows that there are flags, and they are not all equal. My guess is that 
* `"1"` means good
* `" "` means possibly bad
* `"4"` means really bad
but I'm guessing that just on the number of each character that I see from the above commands.

**Comment by dankelley** [comment]
OK, looked up http://www.argodatamgt.org/content/download/12096/80327/file/argo-dm-user-manual.pdf and ref table 2 helps. But they don't seem to say what a blank code means. @richardsc -- any idea?

**Comment by dankelley** [comment]
OK, maybe I get it now. The flag is blank if the item is NA. I infer that from below.
```r
> pflag[14]
[1] "4444444444444444444444444444444444444444444444444444    "
> argo@data$pressure[,14]
 [1]    4.3    9.6   14.3   19.2   24.0   29.1   34.4   39.1   49.3   59.5   69.5   79.3   89.0   99.4  109.1  119.1  129.4  139.0  149.1  159.4  169.5
[22]  179.5  189.2  199.1  219.2  239.6  259.6  279.1  299.1  329.0  359.3  399.4  448.7  499.2  549.5  599.1  649.2  699.3  848.8  899.2  949.4  999.1
[43] 1049.2 1099.3 1149.1 1198.9 1248.9 1299.5 1349.5 1383.0 3411.6 6178.5     NA     NA     NA     NA
```

**Comment by dankelley** [comment]
From all of the above, I feel that I've made some progress on a *new* issue. So I'm going to open one and cut/paste a lot of these comments there (and delete them from here, probably, since I hate scrolling in GH) and then return to the present task.

**Comment by dankelley** [comment]
Note to @richardsc -- I deleted a bunch of comments after I copied them over to #829. I'll delete this one too, in a while. I know it's super-annoying if you're replying using the email interface, but I find it hard to handle GH issues that cross a lot of screenfuls and there seems no need for that, for back-and-forth conversations (including working notes, which is what the last half-dozen comments of mine were).

**Comment by dankelley** [comment]
I think this is working in `develop` commit 47d59895a155e90a63321efd7bbc01765e1b4e69. I added a test in oce-issues.  Sorry for the delay.

**Comment by richardsc** [comment]
> @dankelley: I'm inclined to just find anything that is over 2km depth and set it to NA. But maybe the data mean something..?? Any ideas or suggestions?

Not sure what this is, but I'll have a look. I doubt that those are actually meaningful data, but I don't like the idea to just setting them to NA since there are new "Deep" Argo floats starting to come online that profile to 4000 dbar.

**Comment by dankelley** [comment]
So sorry, Clark, but this has been answered already in #829 .

I deleted a bunch of comments that were sort of notes to myself.  I feel that the present issue as stated has been addressed but I'll let you decide.

I'm sorry about all the "at"s today. I didn't realize that I would make so much progress by twiddling, but I think I've basically got flags working, and the present issue fixed, today.  Will be offline until tomorrow. Dan.
